### PR TITLE
fix(git-semver-tags): escape regexp characters in the tagPrefix option

### PR DESCRIPTION
**Pull request by mattlewis92**
Currently if you're using a `+` symbol in the tagPrefix option it doesn't work as the character isn't escaped when passed to the regexp. This fixes it and adds a test to verify it works correctly.

### CODE REVIEW

File: packages/git-semver-tags/test/index.spec.js [hunk 1]
```diff
@@ -257,6 +257,24 @@ describe('git-semver-tags', () => {
...
+    exec('git add --all && git commit -m"eleventh commit"')
+    exec('git tag notms+7.0.0')
+
+    gitSemverTags({ tagPrefix: 'ms+' }, function (err, tags) {
+      if (err) done(err)
+      assert.deepStrictEqual(tags, ['ms+7.0.0', 'ms+6.0.0'])
+      done()
+    })
+  })
+
```
Comment (dangreen) [inline packages/git-semver-tags/test/index.spec.js:277]: Please see how other tests are written. You should use promises, `testTools` and `expect` util.

File: packages/git-semver-tags/index.js [hunk 2]
```diff
@@ -36,7 +37,7 @@ module.exports = function gitSemverTags (opts, callback) {
     const tags = []
     let tagPrefixRegexp
     if (options.tagPrefix) {
-      tagPrefixRegexp = new RegExp('^' + options.tagPrefix + '(.*)')
+      tagPrefixRegexp = new RegExp('^' + escapeStringRegexp(options.tagPrefix) + '(.*)')
```
[the remark this answers is not in the document]
Comment (mattlewis92) [reply packages/git-semver-tags/index.js]: Great point, done!

File: packages/git-semver-tags/test/index.spec.js (hunk 1 above)
Comment (mattlewis92) [reply packages/git-semver-tags/test/index.spec.js]: oops, missed the new test format when syncing in the latest master, fixed!

**Comment by mattlewis92** [comment]
@dangreen all changed addressed, thanks for the great feedback!

**Review by dangreen — APPROVED** [review]

**Pull request merged**